clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/LeaderFollowerPartitionStateModel.java

```diff
@@ -127,7 +133,18 @@ public void onBecomeOfflineFromStandby(Message message, NotificationContext cont
 
   @Transition(to = HelixState.DROPPED_STATE, from = HelixState.OFFLINE_STATE)
   public void onBecomeDroppedFromOffline(Message message, NotificationContext context) {
-    executeStateTransition(message, context, this::removePartitionFromStoreGracefully);
+    executeStateTransition(message, context, () -> {
+      try {
+        this.threadPoolStats.incrementThreadBlockedOnOfflineToDroppedTransitionCount();
+        // Gracefully drop partition to drain the requests to this partition
+        Thread.sleep(TimeUnit.SECONDS.toMillis(getStoreConfig().getPartitionGracefulDropDelaySeconds()));
```

Comment:
As discussed in the slack channel, this sleep is used to drain the read request.
So the sleep can be skipped if the resource is not for the current version, right?

Reply:
Right; I forgot about this. Will add the check